guides/Other Subjects/CreatingPointsSystems.md

```diff
@@ -0,0 +1,161 @@
+
+Now that we have set up the schema, we will want to create a monitor
+
+```javascript
+const { Monitor } = require('klasa');
+
+module.exports = class extends Monitor {
+
+	constructor(...args) {
+		super(...args, {
+			enabled: true,
+			ignoreBots: true,
+			ignoreSelf: true,
+			ignoreOthers: false
+		});
+	}
+
+	async run(msg) {
+		// If the message was not sent in a TextChannel, ignore it.
+		if (!msg.guild) return;
+
+		// Update the user's configuration entry by adding 1 to it.
+		await msg.author.configs.update('experience', msg.author.configs.experience + 1);
+	}
+
+};
+
+```
+
+Alternatively, we can create the `init` method and ensure the users' schema always has our key.
+
+## Level up!
+
+Some social bots have level up messages, but, how do we set it up? There are two ways to achieve this:
+
+1. We calculate the current level and the next level on the fly. This system is, however, harder to implement and it processes a lot of maths, but it's also RAM friendly for massive bots. We won't cover this in this guide.
+1. We add a level field. This makes the configuration update slower by nature as it will need to update two values. First, we will create the key:
+
+```javascript
+async function init() {
+	if (!this.client.gateways.users.schema.has('level')) {
+		this.client.gateways.users.schema.add('level', {
+			type: 'integer',
+			default: 0,
+			configurable: false
+		});
+	}
+}
+```
+
+Then we pick up a level calculation algorithm, the following as an example:
+
+```javascript
+Math.floor(0.1 * Math.sqrt(POINTS + 1));
+```
+
+Then inside our monitor's run method:
+
+```javascript
+class extends Monitor {
+
+	// Constructor
+
+	async run(msg) {
+		// If the message was not sent in a TextChannel, ignore it.
+		if (!msg.guild) return;
+
+		// Calculate the next value for experience.
+		const nextValue = msg.author.configs.experience + 1;
+
+		// Cache the current level.
+		const currLevel = msg.author.configs.level;
+
+		// Calculate the next level.
+		const nextLevel = Math.floor(0.1 * Math.sqrt(nextValue + 1));
+
+		// Update the user's configuration entry by adding 1 to it, and update the level aswell.
+		await msg.author.configs.update(['experience', 'level'], [nextValue, nextLevel]);
+
+		// If the current level and the next level do not equal, then it has increased, and you can send the message.
+		if (currLevel !== nextLevel) {
+			// Send the message to the channel congratulating the user.
+			await msg.send(`Congratulations! You leveled up to level **${currLevel}**!`);
+		}
+	}
+
+	// Init
+
+}
+```
+
+Optionally, you can check if `nextLevel === msg.author.configs.level` is true and update a single key instead, but the speed difference is negligible and since [SettingGateway v2.1](https://github.com/dirigeants/klasa/pull/179), the key `level` will not be updated if it did not change. As well as this overload is much faster than the JSON object overload, previously used as the only way to update multiple values.
```

Comment:
"As well as this overload is much faster than..." would be better worded as: "As well, this overload is much faster".